test/integration/good/code-gen/cpp.expected

```diff
@@ -12030,6 +12030,887 @@ stan::model::model_base& new_model(
+      } else {
+        current_statement__ = 2;
+        y = stan::math::lb_constrain(y, 0);
+      }
+      {
+        current_statement__ = 3;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 4;
+        if (logical_lt(x, 0.0)) {
+          current_statement__ = 4;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 4;
+          lp_accum__.add(-normal_ccdf_log(0.0, m, 1));
+        }
+        current_statement__ = 5;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 6;
+        if (logical_gt(x, 10.0)) {
+          current_statement__ = 6;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 6;
+          lp_accum__.add(-normal_cdf_log(10.0, m, 1));
+        }
+        current_statement__ = 7;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 9;
+        if (logical_lt(x, 0.0)) {
+          current_statement__ = 9;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 8;
+          if (logical_gt(x, 10.0)) {
+            current_statement__ = 8;
+            lp_accum__.add(stan::math::negative_infinity());
+          } else {
+            current_statement__ = 8;
+            lp_accum__.add(
+              -log_diff_exp(normal_cdf_log(10.0, m, 1),
+                 normal_cdf_log(0.0, m, 1)));
+          }
+        }
+        current_statement__ = 10;
+        lp_accum__.add(normal_log<propto__>(x, m, 1));
+        current_statement__ = 11;
+        lp_accum__.add(poisson_log<propto__>(n, y));
+        current_statement__ = 12;
+        if (logical_lt(n, 10)) {
+          current_statement__ = 12;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 12;
+          lp_accum__.add(-poisson_ccdf_log((10 - 1), y));
+        }
+        current_statement__ = 13;
+        lp_accum__.add(poisson_log<propto__>(n, y));
+        current_statement__ = 14;
+        if (logical_gt(n, 20)) {
+          current_statement__ = 14;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 14;
+          lp_accum__.add(-poisson_cdf_log(20, y));
+        }
+        current_statement__ = 15;
+        lp_accum__.add(poisson_log<propto__>(n, y));
+        current_statement__ = 17;
+        if (logical_lt(n, 10)) {
+          current_statement__ = 17;
+          lp_accum__.add(stan::math::negative_infinity());
+        } else {
+          current_statement__ = 16;
+          if (logical_gt(n, 20)) {
+            current_statement__ = 16;
+            lp_accum__.add(stan::math::negative_infinity());
+          } else {
+            current_statement__ = 16;
+            lp_accum__.add(
+              -log_diff_exp(poisson_cdf_log(20, y),
```

Comment:
Good for line 17 of source with both difference, sign and bounds adjustment for CCDF.